Sources/Authentication/AppAuthSession.swift

```diff
@@ -70,16 +72,20 @@ public final class AppAuthSession: LoginSession {
         }
     }
     
-    
+    /// Finishes the login dialog
+    ///
+    /// - Parameters:
+    ///     - callback: the URL to an authorization endpoint you started the flow with.
+    ///     - endpoint: the token endpoint to query to receive tokens back as the final action in the flow.
     @MainActor
-    public func finalise(callback url: URL) async throws -> TokenResponse {
+    public func finalise(callback url: URL, endpoint: URL) async throws -> TokenResponse {
         flow?.resumeExternalUserAgentFlow(with: url)
         
         guard let authorizationCode else {
             throw LoginError.inconsistentStateResponse
         }
         return try await service
-            .fetchTokens(authorizationCode: authorizationCode)
+            .fetchTokens(authorizationCode: authorizationCode, redirectURI: redirectURI!, endpoint: endpoint)
```

Comment:
a `guard let` plus throwing an error might be better than force unwrapping the string here, we probably don't want the app to crash if the value is nil

Reply:
Yep, definitely, was trying to a the quick-and-dirty implementation in to test the call to token endpoint 😅
Will get that changed 👍